File: client/src/types/stock_deposito.ts
import type { IProductoSimple } from "./producto";
import type { IDepositoSimple } from "./deposito";

export interface IResponseStockDeposito {
    id: number;
    nombre: string;
    descripcion: string;
    producto: IProductoSimple; // id y nombre
    deposito: IDepositoSimple; // id y nombre
    cantidad_producto: number;

}

export interface ICreateStockDeposito {
    nombre: string;
    descripcion: string;
    id_producto: number;
    id_deposito: number;
    cantidad_producto: number;
}


export interface IStockDepositoPatch {
    id_producto: number | null;
    id_deposito: number | null;
    cantidad_producto: number | null;
    nombre: string | null;
    descripcion: string | null;
}

export interface IStockDepositoUpdate {
    id_producto: number;
    id_deposito: number;
    cantidad_producto: number;
    nombre: string;
    descripcion: string;
}

File: src/app/models/stock_deposito.py
from sqlalchemy import Column, DateTime, Float, ForeignKey, Integer, String
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func

from app.db import Base


class StockDeposito(Base):
    __tablename__ = "stock_deposito"
    id = Column(Integer, primary_key=True, index=True)
    id_producto = Column(Integer, ForeignKey("productos.id"), nullable=False)
    id_deposito = Column(Integer, ForeignKey("deposito.id"), nullable=False)
    nombre = Column(String(255), nullable=False)
    descripcion = Column(String(255), nullable=True)
    cantidad_producto = Column(Float, nullable=False, default=0) 
    actualizado_en = Column(DateTime, nullable=False, default=func.now())
    #relaciones
    producto = relationship("Productos")
    deposito = relationship("Deposito")


